### PR TITLE
Added shawnlauzon's fix for #52, which he reported in #53

### PR DESCRIPTION
**Pull request by kenglxn**
This is a cherry pick of his fix commit #15cfb0d7.
Verified that his changes fixes the issue

### CODE REVIEW

File: src/main/groovy/com/jvoegele/gradle/plugins/android/AndroidSetupFactory.groovy [hunk 1]
```diff
@@ -44,7 +44,9 @@ class AndroidSetupFactory {
   AndroidSetup getAndroidSetup() {
 	if (this.androidSdkToolsRevision < 14) {
 	  return new AndroidSetup_r13(project)
-	} else {
+	} else if (this.androidSdkToolsRevision >= 17) {
```
Comment (ealden) [inline src/main/groovy/com/jvoegele/gradle/plugins/android/AndroidSetupFactory.groovy]: While this works, would it be possible to follow the flow of the if-else statement to:

``` groovy
if (this.androidSdkToolsRevision < 14) {
  return new AndroidSetup_r13(project)
} else if (this.androidSdkToolsRevision < 17) {
  return new AndroidSetup_r14(project)
} else {
  return new AndroidSetup_r17(project)
}
```

Should make this much easier to understand.

**Comment by ealden** [comment]
@kenglxn @shawnlauzon I'm getting a compile error with this PR, which I would think is related to the `AndroidSetup_r17` class.  I haven't done a comparison on what exactly changed, but it compiles fine in master.  Exact error is:

```
21:13:35.635 [ERROR] [org.gradle.BuildExceptionReporter] Caused by: : Forked groovyc returned error code: 1
21:13:35.637 [ERROR] [org.gradle.BuildExceptionReporter]    at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
21:13:35.639 [ERROR] [org.gradle.BuildExceptionReporter]    at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
21:13:35.640 [ERROR] [org.gradle.BuildExceptionReporter]    at org.gradle.api.internal.project.ant.BasicAntBuilder.nodeCompleted(BasicAntBuilder.java:71)
21:13:35.640 [ERROR] [org.gradle.BuildExceptionReporter]    at org.gradle.api.internal.project.AntBuilderDelegate.nodeCompleted(DefaultIsolatedAntBuilder.groovy:173)
21:13:35.641 [ERROR] [org.gradle.BuildExceptionReporter]    at org.gradle.api.internal.tasks.compile.AntGroovyCompiler$_execute_closure1.doCall(AntGroovyCompiler.groovy:60)
```

Are you guys encountering the same thing?

**Comment by kenglxn** [comment]
That looks exactly like the error I was getting before the fix.

Are you sure that the new version of the plugin is being called?
I ended up deploying a new version of the plugin to my local mvn repo and depending on it in my build script:

```
buildscript {
  repositories {
    mavenLocal()
    mavenCentral()
  }
  dependencies {
    classpath 'org.gradle.api.plugins:gradle-android-plugin:1.1.1-SNAPSHOT'
  }
}
```

**Comment by ealden** [comment]
@kenglxn I'm actually just running `./gradlew developerBuild`.

Is your master where you built the SNAPSHOT updated?  Also, what Gradle version are you using?

**Comment by kenglxn** [comment]
@ealden Yes, up to date on master: 

```
 ~/dev/git/gradle-android-plugin (master) $ git pull
Already up-to-date.
 ~/dev/git/gradle-android-plugin (master) $ gradle -v 

------------------------------------------------------------
Gradle 1.0-milestone-9
------------------------------------------------------------

Gradle build time: Tuesday, March 13, 2012 4:10:09 PM UTC
Groovy: 1.8.6
Ant: Apache Ant(TM) version 1.8.2 compiled on December 20 2010
Ivy: 2.2.0
JVM: 1.6.0_29 (Apple Inc. 20.4-b02-402)
OS: Mac OS X 10.7.3 x86_64
```

**Comment by ealden** [comment]
@kenglxn okay I removed the `build/` and `out/` folders and I could compile again :).  Wonder what happened there.

In any case this seems to work with R17.  If you could modify `AndroidSetupFactory` with my comment I'll be happy to merge this in.

**Comment by kenglxn** [comment]
When I try ./gradlew developerBuild i get:

```
Unzipping /Users/kenglxn/.gradle/wrapper/dists/gradle-1.0-milestone-8a-bin/5ok1i5pfhkp1uo5oljn22k4kl8/gradle-1.0-milestone-8a-bin.zip to /Users/kenglxn/.gradle/wrapper/dists/gradle-1.0-milestone-8a-bin/5ok1i5pfhkp1uo5oljn22k4kl8
Exception in thread "main" java.util.zip.ZipException: error in opening zip file
    at java.util.zip.ZipFile.open(Native Method)
    at java.util.zip.ZipFile.<init>(ZipFile.java:127)
    at java.util.zip.ZipFile.<init>(ZipFile.java:143)
    at org.gradle.wrapper.Install.unzip(Install.java:148)
    at org.gradle.wrapper.Install.createDist(Install.java:65)
    at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:129)
    at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:47)
```

**Comment by ealden** [comment]
@kenglxn I'm not sure if you've resolved your zip error but perhaps the download did not complete and you ended up with a corrupted zip file.  You can simply remove it in the wrapper/dists directory and do a `./gradlew wrapper` to redownload.

Gradle Android Plugin doesn't work with 1.0-milestone-9 at the moment :(

**Comment by ealden** [comment]
@kenglxn looks good!  Thanks to you and to @shawnlauzon for the contribution!

**Pull request merged**